sdk/storage/azure-storage-file-datalake/out/azure/storage/filedatalake/_file_system_client.pyi

```diff
@@ -0,0 +1,33 @@
+# Stubs for azure.storage.filedatalake._file_system_client (Python 3)
+#
+# NOTE: This dynamically typed stub was automatically generated by stubgen.
+
+from ._data_lake_directory_client import DataLakeDirectoryClient
+from ._data_lake_file_client import DataLakeFileClient
+from ._data_lake_lease import DataLakeLeaseClient
+from ._generated import DataLakeStorageClient
+from ._models import FileSystemProperties, LocationMode, PathPropertiesPaged
+from ._serialize import convert_dfs_url_to_blob_url
+from ._shared.base_client import StorageAccountHostsMixin, parse_connection_str, parse_query
+from azure.core.paging import ItemPaged
+from typing import Any, Optional
+
+class FileSystemClient(StorageAccountHostsMixin):
+    file_system_name: Any = ...
+    def __init__(self, account_url: str, file_system_name: str, credential: Optional[Any]=..., **kwargs: Any) -> None: ...
+    @classmethod
+    def from_connection_string(cls: Any, conn_str: str, file_system_name: str, credential: Optional[Any]=..., **kwargs: Any) -> FileSystemClient: ...
+    def acquire_lease(self, lease_duration: int=..., lease_id: Optional[str]=..., **kwargs: Any) -> DataLakeLeaseClient: ...
+    def create_file_system(self, metadata: Optional[Dict[str, str]]=..., public_access: Optional[PublicAccess]=..., **kwargs: Any) -> Dict[str, Union[str, datetime]]: ...
```

Comment:
We should support `Optional[Union[str, PublicAccess]]` for the `public_access` type.

Reply:
ah, I didn't add a str type because PublicAccess.File is actually 'blob' 😄 but if we really want to support str type we can have a pr to enable that.

Reply:
Not necessary right now - we can add later if need be :)